src/main/java/com/spingboot/demo/service/FileReadService.java

```diff
@@ -0,0 +1,33 @@
+package com.spingboot.demo.service;
+
+import com.spingboot.demo.exceptions.FileProcessingException;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FileReadService {
```

Comment:
let's have some abstract interface and make this class implement it

this class is designed to read file from the file system - so we can name it `SystemFileReaderService implements FileReaderService` for example